query/query.go

```diff
@@ -2383,6 +2398,31 @@ func ProcessGraph(ctx context.Context, sg, parent *SubGraph, rch chan error) {
 	rch <- childErr
 }
 
+// applies "every" to sg.uidMatrix[*].Uids
+// every 'every-th' uid is selected from the uid lists,
+func (sg *SubGraph) applyEvery(ctx context.Context) error {
+	sg.updateUidMatrix()
+
+	every := sg.Params.Every
+	for i := 0; i < len(sg.uidMatrix); i++ {
+		uidList := sg.uidMatrix[i].Uids
+
+		// take every 'every'-th uid
+		uidsLen := len(uidList) / every
+		uids := make([]uint64, uidsLen)
+		for j := 0; j < uidsLen; j++ {
+			// take the last uid of every-sized batches
+			// taking the last rather than the first allows us to paginate via after
+			uids[j] = uidList[(j+1)*every-1]
+		}
+		sg.uidMatrix[i].Uids = uids
+	}
+
+	// Re-merge the UID matrix.
+	sg.DestUIDs = algo.MergeSorted(sg.uidMatrix)
```

Comment:
Can you also write unit test for applyEvery function? 
I think you need to first MergeSort  the entire uidMatrix and then take every ith element? Right now I don't think we will get the correct result.